test/loopback.test.js

```diff
@@ -853,6 +852,7 @@ describe('loopback', function() {
           },
         },
       });
+      app.model(RelatedModel, {dataSource: 'test'});
```

Comment:
I think we should have two tests: one checking the execution path where the related model has been already configured, the second checking the case when the related model is configured later.